Artifacts/windows-visualstudio/artifact.ps1

```diff
@@ -0,0 +1,122 @@
+Param(
+    [Parameter(Mandatory=$true)]
+    [ValidateSet("2015","2017")] 
+    [string] $version,
+
+    [Parameter(Mandatory=$true)]
+    [ValidateSet("Professional","Enterprise")] 
+    [string] $sku,
+
+    [Parameter()]
+    [string] $installerArgs
+)
+
+function DownloadToFilePath ($downloadUrl, $targetFile)
+{
+    Write-Output ("Downloading installation files from URL: $downloadUrl to $targetFile")
+    $targetFolder = Split-Path $targetFile
+
+    if((Test-Path -path $targetFile))
+    {
+        Write-Output "Deleting old target file $targetFile"
+        Remove-Item $targetFile -Force | Out-Null
+    }
+
+    if((Test-Path -path $targetFolder) -eq $false)
+    {
+        Write-Output "Creating folder $targetFolder"
+        New-Item -ItemType Directory -Force -Path $targetFolder | Out-Null
+    }
+
+    #Download the file
+    $downloadAttempts = 0
+    do
+    {
+        $downloadAttempts++
+
+        try
+        {
+            $WebClient = New-Object System.Net.WebClient
+            $WebClient.DownloadFile($downloadUrl,$targetFile)
+            break
+        }
+        catch [Exception]
+        {
+            Write-Output "Caught exception during download..."
+            if ($_.Exception.InnerException){
+                $exceptionMessage = $_.InnerException.Message
+                Write-Output "InnerException: $exceptionMessage"
+            }
+            else {
+                $exceptionMessage = $_.Message
```

Comment:
Are you sure this is not supposed to be `$_.Exception.Message`?

Reply:
You're right.